src/Altinn.App.Core/Implementation/DefaultTaskEvents.cs

```diff
@@ -114,16 +117,15 @@ public async Task OnEndProcessTask(string endEvent, Instance instance)
         Guid instanceGuid = Guid.Parse(instance.Id.Split("/")[1]);
         foreach (DataType dataType in dataTypesToLock)
         {
-            bool generatePdf = dataType.AppLogic?.ClassRef != null && dataType.EnablePdfCreation;
-
             foreach (DataElement dataElement in instance.Data.FindAll(de => de.DataType == dataType.Id))
             {
                 dataElement.Locked = true;
                 _logger.LogInformation($"Locking data element {dataElement.Id} of dataType {dataType.Id}.");
                 Task updateData = _dataClient.Update(instance, dataElement);
 
-                if (generatePdf)
+                if (dataType.AppLogic?.ClassRef != null && dataType.EnablePdfCreation)
```

Comment:
Not sure how the compiler would actually do here, but are we potentially evaluating this boolean for each DataElement for each DataType instead of just for each DataType?

Reply:
Yes, but Altinn has way more important optimizations that could be done that would actually improve something.

I agree this looks strange, but it was the simplest way to get inference to work for nullability checks. An obvious alternative would be to add ‘.!’ to suppress the warnings.

Reply:
Shouldn't we opt for the `!.` approach? Then we would not introduces another, although small, performance reduction and the code should still be null safe as the if statement still checks that `dataType.AppLogic?.ClassRef` isn't null?

Reply:
Both approaches works, but I find it hard to reason about .! when the condition is far away as in this case.

Reply:
I'm disapointed that the analyser is unable to see, through the original boolean variable, that the variable cannot be null when used further down.

Reply:
I agree that it's hard to reason about !. when the condition is far away, but then again the if statement is harder to read with these changes, in my opinion. I vote for the !. approach since this make the code as a whole more readable and we avoid a small performance hit. 

Maybe we should file a featurerequest/bug to the compiler team? 😄

Reply:
I have reverted this change. I'd really like the to get rid of the other almost 200 warnings.